src/core/packages/elasticsearch/client-server-internal/src/agent_manager.ts

```diff
@@ -68,6 +69,57 @@ export class AgentManager implements AgentFactoryProvider, AgentStatsProvider {
         maxTtl: options.dnsCacheTtlInSeconds,
       });
     }
+
+    this.registerMetrics();
+  }
+
+  private registerMetrics() {
+    const meter = metrics.getMeter('kibana.elasticsearch.client');
+
+    const totalActiveSocketsObservable = meter.createObservableGauge(
+      'elasticsearch.client.total_active_sockets',
+      {
+        description:
+          'Elasticsearch Clients: Total number of active sockets (all nodes, all connections)',
+        unit: '1',
+        valueType: ValueType.INT,
+      }
+    );
+    const totalIdleSocketsObservable = meter.createObservableGauge(
+      'elasticsearch.client.total_idle_sockets',
+      {
+        description:
+          'Elasticsearch Clients: Total number of available sockets (alive but idle, all nodes, all connections)',
+        unit: '1',
+        valueType: ValueType.INT,
+      }
+    );
+    const totalQueuedRequestsObservable = meter.createObservableGauge(
+      'elasticsearch.client.total_queued_requests',
+      {
+        description:
+          'Elasticsearch Clients: Total number of queued requests (all nodes, all connections)',
+        unit: '1',
+        valueType: ValueType.INT,
+      }
+    );
+
+    meter.addBatchObservableCallback(
```

Comment:
I could have instrumented it like below, but that would require looping throught he agents 3 times. That's why I went with `addBatchObservableCallback`.

```js
totalActiveSocketsObservable.addCallback((result) => {
  let totalActiveSockets = 0;
  this.agents.forEach((agent) => {
    totalActiveSockets += Object.values(agent.sockets).flat().length;
  });
  result.observe(totalActiveSockets);
});
```

Reply:
Also, we have to use the observable methods here because the HTTP agents do not expose any events to track the creation of sockets (from a top level, we can do `agent.get().on('socket', ...)`)